### PR TITLE
Add onSelectedRangeChange

### PR DESCRIPTION
**Pull request by ingun37**
## Description

To what I understood, which can be wrong, the UITextView/NSTextView is hidden therefore it's impossible to know where the cursor is within the document. So I exposed both bindable for selection range value.

## Example

It's working for my iOS application and `SwiftDownDemo`.

Feel free to close or request changes. I'm happy to contribute.

### CODE REVIEW

**Comment by ingun37** [comment]
I'm sorry I managed to run `SwiftDownDemo` and found out it's not working on macOS. I'm working on it.

**Comment by ingun37** [comment]
Fixed

**Comment by qeude** [comment]
Hey @ingun37,
Thanks for your contributions, I'm currently moving from one location to another so I might not be able to take a closer look right now. But I will as soon as possible!

File: SwiftDownDemo/Shared/ContentView.swift [hunk 1]
```diff
@@ -16,16 +16,20 @@ struct ContentView: View {
...
       .focused($focusedField, equals: .field)
       .onAppear {
         self.focusedField = .field
       }
       TextEditor(text: $text)
-      Button {
-        self.text = ""
-      } label: {
-        Text("Clear")
+      HStack {
```
Comment (qeude) [inline SwiftDownDemo/Shared/ContentView.swift]: Why this `HStack`?

**Review by qeude — APPROVED** [review]
Thanks again for your contribution, this is something I saw back then but didn't take time to fix it.

Overall LGTM, just a question about the addition of a `HStack`

**Comment by ingun37** [comment]
@qeude This one I'll finish soon

**Comment by ingun37** [comment]
I'm having trouble setting up my new laptop. I know I can just edit from GitHub since it's a very simple fix but I just want to run it before merge just in case. So if it's not in hurry please wait.

**Comment by qeude** [comment]
Hi @ingun37,
No worries, there is no rush on this 🙏 Take your time and do this when you are ready to do so 😄

**Comment by ingun37** [comment]
@qeude Removed HStack. Please review again.

**Pull request merged**